From 5830b090a6c62ab216dbd378c05f489d0c5fae10 Mon Sep 17 00:00:00 2001
From: Silin Huang <silin@google.com>
Date: Mon, 1 May 2023 17:59:15 +0000
Subject: [PATCH] Do not load drawable for wallet card if the card image icon
 is created with content URI.

This prevents the primary user from accessing the secondary user's
photos for QAW card images.

Test: manual
Bug: 272020068
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:36284852414f25d2977100fc98d1b1db4f7e6482)
Merged-In: Ie02f028e5b937a1887239839f59313ee0509337b
Change-Id: Ie02f028e5b937a1887239839f59313ee0509337b
---
 .../wallet/WalletPanelViewController.java              | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/com/android/systemui/plugin/globalactions/wallet/WalletPanelViewController.java b/src/com/android/systemui/plugin/globalactions/wallet/WalletPanelViewController.java
index 22165d4..4e9ce4f 100644
--- a/src/com/android/systemui/plugin/globalactions/wallet/WalletPanelViewController.java
+++ b/src/com/android/systemui/plugin/globalactions/wallet/WalletPanelViewController.java
@@ -24,6 +24,7 @@ import android.content.res.Resources;
 import android.graphics.drawable.Drawable;
 import android.graphics.drawable.Icon;
 import android.os.Handler;
+import android.os.UserHandle;
 import android.os.Looper;
 import android.service.quickaccesswallet.GetWalletCardsError;
 import android.service.quickaccesswallet.GetWalletCardsRequest;
@@ -361,7 +362,14 @@ public class WalletPanelViewController implements
          */
         QAWalletCardViewInfo(WalletCard walletCard) {
             mWalletCard = walletCard;
-            mCardDrawable = mWalletCard.getCardImage().loadDrawable(mPluginContext);
+            Icon cardImage = mWalletCard.getCardImage();
+            if (cardImage.getType() == Icon.TYPE_URI) {
+                // Do not allow icon created with content URI.
+                mCardDrawable = null;
+            } else {
+                mCardDrawable =
+                    mWalletCard.getCardImage().loadDrawable(mPluginContext);
+            }
             Icon icon = mWalletCard.getCardIcon();
             mIconDrawable = icon == null ? null : icon.loadDrawable(mPluginContext);
         }
-- 
2.41.0.255.g8b1d071c50-goog

